internal/cmd/controller/agentmanagement/controllers/cluster/import_test.go

```diff
@@ -89,39 +89,79 @@ func TestOnConfig(t *testing.T) {
 				AgentTLSMode:              "system-store",
 				GarbageCollectionInterval: metav1.Duration{Duration: 10 * time.Minute},
 			},
-			handlerWithMocks: func() importHandler {
+			handlerWithMocks: func(t *testing.T) importHandler {
 				ctrl := gomock.NewController(t)
 
 				secretsCache := fake.NewMockCacheInterface[*corev1.Secret](ctrl)
 				secretsCache.EXPECT().Get(gomock.Any(), "my-kubeconfig-secret").Return(&corev1.Secret{}, nil)
 
-				clustersController := fake.NewMockControllerInterface[*fleet.Cluster, *fleet.ClusterList](ctrl)
-				clustersController.EXPECT().List("", metav1.ListOptions{}).
-					Return(&fleet.ClusterList{
-						Items: []fleet.Cluster{
-							{
-								ObjectMeta: metav1.ObjectMeta{
-									Name:      "cluster",
-									Namespace: "fleet-default",
-								},
-								Spec: fleet.ClusterSpec{
-									KubeConfigSecret: "my-kubeconfig-secret",
-								},
-								Status: fleet.ClusterStatus{
-									APIServerURL:              "https://hello.world",
-									APIServerCAHash:           hashStatusField("foo"),
-									AgentTLSMode:              "system-store",
-									GarbageCollectionInterval: &metav1.Duration{Duration: 10 * time.Minute},
-								},
+				clustersCache := fake.NewMockCacheInterface[*fleet.Cluster](ctrl)
+				clustersCache.EXPECT().List("", gomock.Eq(labels.Everything())).
+					Return([]*fleet.Cluster{
+						{
+							ObjectMeta: metav1.ObjectMeta{
+								Name:      "cluster",
+								Namespace: "fleet-default",
+							},
+							Spec: fleet.ClusterSpec{
+								KubeConfigSecret: "my-kubeconfig-secret",
+							},
+							Status: fleet.ClusterStatus{
+								APIServerURL:              "https://hello.world",
+								APIServerCAHash:           hashStatusField("foo"),
+								AgentTLSMode:              "system-store",
+								GarbageCollectionInterval: &metav1.Duration{Duration: 10 * time.Minute},
 							},
 						},
 					}, nil)
-
+				clustersController := fake.NewMockControllerInterface[*fleet.Cluster, *fleet.ClusterList](ctrl)
 				clustersController.EXPECT().UpdateStatus(gomock.Any()) // import triggered
 
 				return importHandler{
-					clusters: clustersController,
-					secrets:  secretsCache,
+					clusters:      clustersController,
+					clustersCache: clustersCache,
+					secretsCache:  secretsCache,
+				}
+			},
+		},
+		"non-ready config and no URL or CA in secret, do not trigger import when CA changes": {
+			cfg: config.Config{
+				APIServerURL:              "",
+				AgentTLSMode:              "system-store",
```

Comment:
nit: this test case could even exhibit a TLS mode update, to demonstrate that other changes would not be taken into account anyway if no API server URL is available:
```suggestion
				AgentTLSMode:              "strict",
```

Reply:
Great point, thanks!